### PR TITLE
Handle popover initialization globally

### PR DESCRIPTION
**Pull request by nwalters512**
This PR started out with the goal of fixing a specific problem I was facing the Bootstrap 5 migration. We have popovers in some dynamically-rendered tables, which required us to run re-initialization code whenever the table was updated. This conflicts with our "rewrite everything in the DOM" strategy for BS5 compat, as the legacy `data-*` attributes (without the `data-bs-` prefix) weren't rewritten in time for that reinitialization process.

The solution is of course to do things more declaratively than imperatively. That is, rather that reacting specifically to table rerenders, we can use [`selector-observer`](https://github.com/josh/selector-observer) to watch for new popover triggers to appear and automatically initialize them. Neat!

I could have done this just on the relevant pages, but it was almost easier to make this change globally, so that's what I did. It's now no longer necessary to sprinkle `$('[data-toggle="popover"]').popover({ sanitize: false });` all over everything.

## TODO

- [x] Some popover initialization had set `container: 'body'`. I have no idea if/when this was necessary. My guess is that it was just copied around all over the place. I need to test basically every use case (😢) to see if it's necessary and replace that with `data-container="body"`.
- [x] Test all elements.

### CODE REVIEW

**Comment by trombonekenny** [comment]
Is it a concern that `selector-observer` is unmaintained?

**Comment by nwalters512** [comment]
> Is it a concern that selector-observer is unmaintained?

No; I'd call it "finished software". We use it extensively in PT. It has a very small API surface area and does exactly what we need it to.

**Comment by nwalters512** [comment]
@mylesw27 tested pretty much every popover we have, and I tested the rest! This appears to be good to go.

File: apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-import { onDocumentReady } from '@prairielearn/browser-utils';
-import './lib/changeIdButton.js';
```
Comment (jonatanschroeder) [inline apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts:2]: Deleting this file is removing functionality from this page, mainly the controls of the "change ID" button.

**Review by eliotwrobson — APPROVED** [review]
Not an expert on how this works, but cuts down on the boilerplate in the element code, looks really good 👍

File: apps/prairielearn/assets/scripts/lib/qrCodeButton.ts [hunk 2]
```diff
@@ -1,19 +1,26 @@
 import QR from 'qrcode-svg';
+import { observe } from 'selector-observer';
 
 import { onDocumentReady, parseHTMLElement } from '@prairielearn/browser-utils';
 
 onDocumentReady(() => {
```
Comment (jonatanschroeder) [inline apps/prairielearn/assets/scripts/lib/qrCodeButton.ts:6]: Nit: with the change to `observe`, `onDocumentReady` is no longer needed.

**Review by mylesw27 — APPROVED** [review]
As @nwalters512  stated, I tested most of these locally. Everything here appears to be good to go!

File: apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts [hunk 3]
```diff
@@ -1,5 +0,0 @@
-import './behaviors/custom-file-input.js';
```
Comment (jonatanschroeder) [inline apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts:1]: Again, removed this functionality.

File: apps/prairielearn/elements/pl-matching/pl-matching.mustache [hunk 4]
```diff
@@ -44,7 +30,7 @@
 <div class="pl-matching-options">
 <ol>
 {{#options}}
-    <li class="pl-matching-option">{{{html}}}</li>
+    <li class="pl-matching-option" style="--pl-matching-counter-type: {{counter_type}}">{{{html}}}</li>
```
Comment (jonatanschroeder) [inline apps/prairielearn/elements/pl-matching/pl-matching.mustache:33]: Clever solution...

**Review by jonatanschroeder — CHANGES_REQUESTED** [review]

File: apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts (hunk 3 above)
Comment (jonatanschroeder) [reply apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts:1]: If you get #10430 out first this may not be a problem...

Comment (nwalters512) [reply apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts:1]: That's exactly why I opened that PR 😅

Comment (nwalters512) [reply apps/prairielearn/assets/scripts/studentAssessmentInstanceClient.ts]: Removed properly via #10430.

File: apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/instructorQuestionSettingsClient.ts:2]: Good catch, restored in 6420857.

File: apps/prairielearn/assets/scripts/lib/qrCodeButton.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/assets/scripts/lib/qrCodeButton.ts]: Removed in a1bebd8.

**Review by jonatanschroeder — APPROVED** [review]

**Pull request merged**